walk-in-interview/src/main/java/com/google/job/data/JobsDatabase.java

```diff
@@ -0,0 +1,70 @@
+package com.google.job.data;
+
+import com.google.api.core.ApiFunction;
+import com.google.api.core.ApiFuture;
+import com.google.api.core.ApiFutures;
+import com.google.cloud.firestore.DocumentReference;
+import com.google.cloud.firestore.DocumentSnapshot;
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.WriteResult;
+import com.google.common.util.concurrent.MoreExecutors;
+import com.google.firestore.FireStoreUtils;
+import org.checkerframework.checker.nullness.compatqual.NullableDecl;
+
+import java.util.concurrent.Future;
+
+/** Helps persist and retrieve Future of job posts. */
```

Comment:
Nit: "Helps persist and retrieve job posts"?. No need the "future" as it's just a means to eventually deliver the job.